ffi/diplomat/src/collator.rs

```diff
@@ -0,0 +1,234 @@
+    pub enum ICU4XOrdering {
+        Less = 0,
+        Equal = 1,
+        Greater = 2,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::CollatorOptions, Struct)]
+    pub struct ICU4XCollatorOptions {
+        pub strength: ICU4XCollatorStrength,
+        pub alternate_handling: ICU4XCollatorAlternateHandling,
+        pub case_first: ICU4XCollatorCaseFirst,
+        pub max_variable: ICU4XCollatorMaxVariable,
+        pub case_level: ICU4XCollatorCaseLevel,
+        pub numeric: ICU4XCollatorNumeric,
+        pub backward_second_level: ICU4XCollatorBackwardSecondLevel,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::Strength, Enum)]
+    pub enum ICU4XCollatorStrength {
+        Auto = 0,
+        Primary = 1,
+        Secondary = 2,
+        Tertiary = 3,
+        Quaternary = 4,
+        Identical = 5,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::AlternateHandling, Enum)]
+    pub enum ICU4XCollatorAlternateHandling {
+        Auto = 0,
+        NonIgnorable = 1,
+        Shifted = 2,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::CaseFirst, Enum)]
+    pub enum ICU4XCollatorCaseFirst {
+        Auto = 0,
+        Off = 1,
+        LowerFirst = 2,
+        UpperFirst = 3,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::MaxVariable, Enum)]
+    pub enum ICU4XCollatorMaxVariable {
+        Auto = 0,
+        Space = 1,
+        Punctuation = 2,
+        Symbol = 3,
+        Currency = 4,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::CaseLevel, Enum)]
+    pub enum ICU4XCollatorCaseLevel {
+        Auto = 0,
+        Off = 1,
+        On = 2,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::Numeric, Enum)]
+    pub enum ICU4XCollatorNumeric {
+        Auto = 0,
+        Off = 1,
+        On = 2,
+    }
+
+    #[non_exhaustive]
+    #[diplomat::rust_link(icu_collator::BackwardSecondLevel, Enum)]
+    pub enum ICU4XCollatorBackwardSecondLevel {
+        Auto = 0,
+        Off = 1,
+        On = 2,
+    }
+
+    impl ICU4XCollator {
+        pub fn try_new(
+            provider: &ICU4XDataProvider,
+            locale: &ICU4XLocale,
+            options: ICU4XCollatorOptions,
+        ) -> DiplomatResult<Box<ICU4XCollator>, ICU4XError> {
+            use icu_provider::serde::AsDeserializingBufferProvider;
+            let provider = provider.0.as_deserializing();
+            let locale = locale.to_datalocale();
+            let options = CollatorOptions::from(options);
+
+            Collator::try_new_unstable(&provider, &locale, options)
+                .map(|o| Box::new(ICU4XCollator(o)))
+                .map_err(Into::into)
+                .into()
+        }
+
+        #[diplomat::rust_link(icu::collator::Collator::compare, FnInStruct)]
+        pub fn compare(&self, left: &str, right: &str) -> ICU4XOrdering {
```

Comment:
nit: probably worth having one line of a doc comment for each one that makes it easy to tell what the difference is at a glance